### PR TITLE
Policies async

### PR DESCRIPTION
**Pull request by AsToNlele**
WIP

## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

File: src/Frameworks/AsyncTableTools/hooks/useAsyncTableTools/helpers.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ export const toToolbarActions = ({
     ? {
         toolbarProps: {
           actionsConfig: {
-            actions: [firstAction, ...actionsOption],
+            actions: [...(firstAction ?? []), ...actionsOption],
```
Comment (mkholjuraev) [inline src/Frameworks/AsyncTableTools/hooks/useAsyncTableTools/helpers.js:11]: Very edge case: what happens if the `firstAction` passed is not iterable? For example if it is a number?

File: src/Frameworks/AsyncTableTools/hooks/useAsyncTableTools/useAsyncTableTools.js [hunk 2]
```diff
@@ -8,6 +8,7 @@ import useExpandable from '../useExpandable';
 import useColumnManager from '../useColumnManager';
 import withExport from '../../utils/withExport';
 import { toToolbarActions } from './helpers';
+import { useActionResolverWithItems } from '../../../../Utilities/hooks/useTableTools/useActionResolver';
```
Comment (mkholjuraev) [inline src/Frameworks/AsyncTableTools/hooks/useAsyncTableTools/useAsyncTableTools.js:11]:
```suggestion
import { useActionResolverWithItems } from '@/Utilities/hooks/useTableTools/useActionResolver';
```

File: src/Frameworks/AsyncTableTools/hooks/useFilterConfig/useFilterConfig.js [hunk 3]
```diff
@@ -61,7 +61,7 @@ const useFilterConfig = (options = {}) => {
   );
 
   useEffect(() => {
-    activeFilters && setTableState(activeFilters);
+    setTableState(activeFilters ?? {});
```
Comment (mkholjuraev) [inline src/Frameworks/AsyncTableTools/hooks/useFilterConfig/useFilterConfig.js:64]: Interesting, why?

File: src/PresentationalComponents/PoliciesTable/PoliciesTable.js [hunk 4]
```diff
@@ -1,13 +1,14 @@
 import React from 'react';
 import propTypes from 'prop-types';
 import { COMPLIANCE_TABLE_DEFAULTS } from '@/constants';
-import { TableToolsTable } from 'Utilities/hooks/useTableTools';
+// import { TableToolsTable } from 'Utilities/hooks/useTableTools';
```
Comment (mkholjuraev) [inline src/PresentationalComponents/PoliciesTable/PoliciesTable.js:4]: Can we remove the commented out line?

File: src/Frameworks/AsyncTableTools/hooks/useAsyncTableTools/useAsyncTableTools.js (hunk 2 above)
Comment (mkholjuraev) [inline src/Frameworks/AsyncTableTools/hooks/useAsyncTableTools/useAsyncTableTools.js:11]: Let's use path alias for imports

```suggestion
import { useActionResolverWithItems } from '@/Utilities/hooks/useTableTools/useActionResolver';
```

File: src/SmartComponents/CompliancePolicies/CompliancePolicies.js [hunk 5]
```diff
@@ -20,9 +19,11 @@ import useAPIV2FeatureFlag from '../../Utilities/hooks/useAPIV2FeatureFlag';
 import PropTypes from 'prop-types';
 import dataSerialiser from '../../Utilities/dataSerialiser';
 import { QUERY, dataMap } from './constants';
+import TableStateProvider from '../../Frameworks/AsyncTableTools/components/TableStateProvider';
```
Comment (mkholjuraev) [inline src/SmartComponents/CompliancePolicies/CompliancePolicies.js:22]: Let's use path alias for imports
```suggestion
import TableStateProvider from '@/Frameworks/AsyncTableTools/components/TableStateProvider';
```

File: src/Utilities/hooks/useAPIV2FeatureFlag.js [hunk 6]
```diff
@@ -1,5 +1,6 @@
 import useFeatureFlag from './useFeatureFlag';
 
-const useAPIV2FeatureFlag = () => useFeatureFlag('compliance-api-v2');
+// const useAPIV2FeatureFlag = () => useFeatureFlag('compliance-api-v2');
+const useAPIV2FeatureFlag = () => true;
```
Comment (mkholjuraev) [inline src/Utilities/hooks/useAPIV2FeatureFlag.js:4]: Let's remove this change

File: src/Utilities/hooks/useHasNoPolicies.js [hunk 7]
```diff
@@ -0,0 +1,9 @@
+import { usePoliciesQuery } from './usePoliciesQuery/usePoliciesQuery';
+
+const useHasNoPolicies = () => {
```
Comment (mkholjuraev) [inline src/Utilities/hooks/useHasNoPolicies.js:3]: Why do we want to make a separate request to check policy presence? Can not we use already fetched policies data in Policies page?

File: src/constants.js [hunk 8]
```diff
@@ -18,9 +18,10 @@ export const API_HEADERS = {
 };
 
 // Add a localStorage entry with the key "insights:compliance:asynctables" and value "true" to enable async tables
-export const ENABLE_ASYNC_TABLE_HOOKS = localStorage
-  ? localStorage.getItem('insights:compliance:asynctables') === 'true'
-  : false;
+// export const ENABLE_ASYNC_TABLE_HOOKS = localStorage
+//   ? localStorage.getItem('insights:compliance:asynctables') === 'true'
+//   : false;
+export const ENABLE_ASYNC_TABLE_HOOKS = true;
```
Comment (mkholjuraev) [inline src/constants.js:24]: hmmm, let's remove this as well.

File: src/Frameworks/AsyncTableTools/hooks/useAsyncTableTools/helpers.js (hunk 1 above)
Comment (bastilian) [reply src/Frameworks/AsyncTableTools/hooks/useAsyncTableTools/helpers.js:11]: I'm interested in what the issue was that prompted this change. We should also add a test for that case.

File: src/Frameworks/AsyncTableTools/hooks/useAsyncTableTools/useAsyncTableTools.js (hunk 2 above)
Comment (bastilian) [reply src/Frameworks/AsyncTableTools/hooks/useAsyncTableTools/useAsyncTableTools.js:11]: Why is the old hook used here?

File: src/Frameworks/AsyncTableTools/hooks/useAsyncTableTools/helpers.js (hunk 1 above)
Comment (AsToNlele) [reply src/Frameworks/AsyncTableTools/hooks/useAsyncTableTools/helpers.js:11]: forgot I could use the firstAction instead, ignore this 💀

File: src/Frameworks/AsyncTableTools/components/AsyncTableToolsTable/AsyncTableToolsTable.js [hunk 9]
```diff
@@ -61,10 +63,17 @@ const AsyncTableToolsTable = ({
     <>
       <PrimaryToolbar {...toolbarProps} />
 
-      <Table {...tableProps}>
-        <TableHeader {...tableHeaderProps} />
-        <TableBody {...tableBodyProps} />
-      </Table>
+      {options.loading ? (
```
[the remark this answers is not in the document]
Comment (bastilian) [reply src/Frameworks/AsyncTableTools/components/AsyncTableToolsTable/AsyncTableToolsTable.js:66]: @AsToNlele The latest `master` should now have a loading state (in the `useItems` hook`) that we could utilise. Let me know if you look into how we could implement a nice loading state with that and how we can integrate it into pages.

**Pull request closed without merging**